src/db.c

```diff
@@ -93,7 +93,7 @@ robj *lookupKey(redisDb *db, robj *key, int flags) {
          * perform additional writes. */
         int is_ro_replica = server.masterhost && server.repl_slave_ro;
         int force_delete_expired = flags & LOOKUP_WRITE && !is_ro_replica;
-        if (expireIfNeeded(db, key, force_delete_expired)) {
+        if (expireIfNeeded(db, key, force_delete_expired, !!(flags&LOOKUP_NOEXPIRE))) {
```

Comment:
The flags are documented in the comment above the function definition of this function `lookupKey`. The new flag should probably be described there.